src/components/CouponDetails/helpers.test.jsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable max-len */
+/* eslint-disable object-curly-newline */
+import {
+  SINGLE_USE, ONCE_PER_CUSTOMER, MULTI_USE, MULTI_USE_PER_CUSTOMER,
+} from '../../data/constants/coupons';
+import { ACTION_LABELS, ACTION_TYPES, COUPON_FILTER_TYPES, FILTER_OPTIONS } from './constants';
+import { getBASelectOptions, getFilterOptions, getFirstNonDisabledOption, shouldShowSelectAllStatusAlert } from './helpers';
+
+describe('getFilterOptions', () => {
+  test.each([
+    [MULTI_USE],
+    [MULTI_USE_PER_CUSTOMER],
+  ])('includes partially redeemed for partially redeemable coupons %p', (usageLimitation) => {
+    expect(getFilterOptions(usageLimitation)).toEqual(FILTER_OPTIONS);
+  });
+  test.each([
+    [SINGLE_USE],
+    [ONCE_PER_CUSTOMER],
+  ])('does not include partially redeemed for single-use coupons %p', (usageLimitation) => {
+    const expected = [{
+      label: 'Unassigned',
+      value: 'unassigned',
+    }, {
+      label: 'Unredeemed',
+      value: 'unredeemed',
+    }, {
+      label: 'Redeemed',
+      value: 'redeemed',
+    }];
+    expect(getFilterOptions(usageLimitation)).toEqual(expected);
+  });
+});
+
+describe('getFirstNonDisabledOption', () => {
+  test.each([
+    [[{ value: 'foo', disabled: true }, { value: 'bar' }], 'bar'],
+    [[{ value: 'foo' }, { value: 'bar', disabled: false }, { value: 'baz', disabled: true }], 'foo'],
```

Comment:
do we expect the potential that more than 1 is not disabled? (I think this is basically the same sort of question I asked before about "first" result)

Reply:
Now that I pulled out the public and private codes, we don't really expect that? But we did at first, and we might again.